### PR TITLE
Update Intel environment

### PR DESCRIPTION
**Pull request by EmilyBourne**
Update the Intel environment to avoid incompatibilities due to different versions for the C/C++ compiler and the Fortran compiler. Create issues and mark all failing tests for Intel. Change the default Intel compilers to the new versions from Intel-OneApi. Intel are deprecating `icc` and `ifort` this year.

### CODE REVIEW

File: tests/epyccel/test_epyccel_mod.py [hunk 1]
```diff
@@ -12,8 +12,8 @@
     RTOL = 1e-13
     ATOL = 1e-14
 elif os.environ.get('PYCCEL_DEFAULT_COMPILER', 'GNU') == 'intel':
-    RTOL = 1e-11
-    ATOL = 1e-14
+    RTOL = 1e-10
+    ATOL = 1e-10
```
Comment (yguclu) [inline tests/epyccel/test_epyccel_mod.py:16]: These tolerances are huge compared to the default ones for GNU. Are we using excessive optimizations with the Intel compilers?

Comment (EmilyBourne) [reply tests/epyccel/test_epyccel_mod.py:16]: Maybe. I'm not sure. I suspect they have a different algorithm as it is only the modulo operator that gives such different values

**Comment by yguclu** [comment]
Is this PR ready for review? 😃

**Review by yguclu — APPROVED** [review]
Looks good!

**Review by yguclu — APPROVED** [review]

**Pull request merged**